### PR TITLE
Added missing EndClassDefinition to ContextAnalyzer

### PR DESCRIPTION
**Pull request by amarekano**

### CODE REVIEW

File: Sources/Fuzzilli/FuzzIL/Analyzer.swift [hunk 1]
```diff
@@ -169,7 +169,8 @@ struct ContextAnalyzer: Analyzer {
     mutating func analyze(_ instr: Instruction) {
```
Comment (saelo) [inline Sources/Fuzzilli/FuzzIL/Analyzer.swift:169]: Given this bug, we should probably have tests for this. Unless you really feel like adding such a test, maybe just add a `// TODO add a test for this analyzer when modifying this function in the future` here?

**Review by saelo — COMMENTED** [review]
Oh good catch, thanks!

File: Sources/Fuzzilli/FuzzIL/Analyzer.swift (hunk 1 above)
Comment (amarekano) [reply Sources/Fuzzilli/FuzzIL/Analyzer.swift:169]: I've added a test for the Context Analyzer

**Comment by amarekano** [comment]
Would we also need additional handling for BeginMethod and EndMethod ops?

File: Sources/Fuzzilli/FuzzIL/Analyzer.swift (hunk 1 above)
Comment (saelo) [reply Sources/Fuzzilli/FuzzIL/Analyzer.swift:169]: thanks!

**Comment by saelo** [comment]
I think we don't, currently. The handler for `BeginClassDefinition` adds the `.function` context, which will stay active throughout the function definition. This works because the content of a `ClassDefinition` only consists of method definitions.

**Comment by amarekano** [comment]
Cool, I guess we're good to merge in that case!

**Pull request merged**